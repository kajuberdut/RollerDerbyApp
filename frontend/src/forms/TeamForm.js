import React, { useState } from "react";
import FastApi from "../Api";
import { useNavigate } from "react-router-dom";
import "./TeamForm.css"
import { Card, CardBody, CardTitle, Form, FormGroup, Label, Input, Button } from "reactstrap";

/** 
 * Form for creating a new team
 */

const TeamForm = () => {

  /** Retrieve user from local storage*/

  const user = JSON.parse(localStorage.getItem('user'));
  const navigate = useNavigate();

  /** Sets Initial State of Form  */

  let INITIAL_STATE = { name: ""};

  /** Sets formData in initial state */

  const [formData, setFormData] = useState(INITIAL_STATE);
  
  /** Handle Submit by creating a team returning an error message */

  const handleSubmit = async evt => {
    evt.preventDefault();   

    setFormData(INITIAL_STATE);
    try {
      let result = await FastApi.addTeam(formData);

      if(result) {
        navigate('/teams')
      }
    } catch(errors) {
      return { success: false, errors };
    }
  }

  /** Update local state with current state of input element */

  const handleChange = evt => {

    const { name, value }= evt.target;

      setFormData(fData => ({
        ...fData,
        [name]: value,
        admin: user.userId,
        type: "team"
        
      }));
  };

  /** render team form */

  return (
    <section className="col-md-4 TeamForm" style={{marginTop: "150px"}}>
        <Card>
            <CardTitle className="TeamForm-CardTitle">
            <h1>Create a Team</h1>
            <p><b>Note: When you create a team you become the admin of that team.</b></p>
            </CardTitle>
            <CardBody>
                <Form onSubmit={handleSubmit}>
                    <FormGroup>
                       
                        <Label htmlFor="name">Name: </Label>
                        <Input
                            type="input"
                            id="name"
                            name="name"
                            value={formData.name}
                            onChange={handleChange}
                            maxLength={50}
                        />
                
                    </FormGroup>
                    <Button >Add Team</Button>
                </Form>
            </CardBody>
        </Card>
    </section> 
  );
};

export default TeamForm;
