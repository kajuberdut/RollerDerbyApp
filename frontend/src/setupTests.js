// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';


global.localStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
};

const user = {
  userId: 1,
  username: "testUser",
  email: "testuser@gmail.com",
  facebookName: "facebook test user",
  about: "This is my story...",
  primaryNumber: "01",
  level: "A",
  ruleset: [],
  position: [],
  insurance: [],
  locationId: null,
  associatedLeagues: "Testing League",
  phoneNumber: "1111111111",
  firstName: "Jane",
  lastName: "Doe",
  additionalInfo: "Epi pen in bag.",
  secondaryNumber: "02"
};

beforeEach(() => {
  localStorage.setItem("user", JSON.stringify(user));
});

