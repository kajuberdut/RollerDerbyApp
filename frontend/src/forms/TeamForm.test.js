import { render, screen, cleanup } from '@testing-library/react';
import TeamForm from './TeamForm';
import { BrowserRouter } from "react-router-dom";
import '../setupTests.js';

// * Passing as of 2/13/24
// npm test TeamForm.test.js
// must be in frontend directory 

afterEach(() => {
    cleanup()
  })
  
describe("checks rendering of team form", () => {  

    test('renders team form page', () => {

    render(
        <BrowserRouter>
                <TeamForm />
        </BrowserRouter>);
    });


    test('matches snapshot', function() {
    const { asFragment } = render(
        <BrowserRouter>
                <TeamForm />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot(); 
    })

});

describe("asserts team form text", () => {  

    test('renders team form page', () => {

    render(
        <BrowserRouter>
                <TeamForm />
        </BrowserRouter>
    )

        const text = screen.getByText('Create a Team');
        expect(text).toBeInTheDocument();
    });

});