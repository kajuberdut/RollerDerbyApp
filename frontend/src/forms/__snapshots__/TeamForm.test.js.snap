// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of team form matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 TeamForm"
    style="margin-top: 150px;"
  >
    <div
      class="card"
    >
      <div
        class="TeamForm-CardTitle card-title"
      >
        <h1>
          Create a Team
        </h1>
        <p>
          <b>
            Note: When you create a team you become the admin of that team.
          </b>
        </p>
      </div>
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="name"
            >
              Name: 
            </label>
            <input
              class="form-control"
              id="name"
              maxlength="50"
              name="name"
              type="input"
              value=""
            />
          </div>
          <button
            class="btn btn-secondary"
          >
            Add Team
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
